addons/web_editor/static/src/components/history_dialog/history_dialog.js

```diff
@@ -0,0 +1,113 @@
+/** @odoo-module **/
+
+import { Dialog } from '@web/core/dialog/dialog';
+import { formatDateTime } from '@web/core/l10n/dates';
+import { useService } from '@web/core/utils/hooks';
+import { memoize } from '@web/core/utils/functions';
+import { Component, onMounted, useState, markup } from '@odoo/owl';
+import { _t } from '@web/core/l10n/translation';
+
+const { DateTime } = luxon;
+
+class HistoryDialog extends Component {
+    static template = 'web_editor.HistoryDialog';
+    static components = { Dialog };
+    static props = {
+        recordId: String,
+        recordModel: String,
+        close: Function,
+        restoreRequested: Function,
+        historyRevisions: Array,
+        versionnedFieldName: String
+    };
+
+    state = useState({
+        revisionsData: [],
+        revisionContent: null,
+        revisionComparison: null,
+        revisionIndex: null
+    });
+
+    setup() {
+        this.size = 'xl';
+        this.title = _t('History');
+        this.orm = useService('orm');
+
+        onMounted(() => this.init());
+    }
+
+    async init() {
+        for (let i = 0; i < this.props.historyRevisions.length; i++) {
+            const rev = this.props.historyRevisions[i];
+            this.props.historyRevisions[i]['user_data'] =
+                await this.getRevisionAuthorData(rev.user_id);
```

Comment:
I'll batch them

Reply:
After Disord discussion : we'll include this data into the `html_field_history_revisions` and `html_field_history_metadata` to avoid an orm call client side altogether.